### PR TITLE
🔥Remove all tracking regarding unrepresentable dates

### LINKED ISSUE #5236: Remove all tracking regarding unrepresentable dates
Once we have fixed all the unrepresentable date errors, there will be no need to track exceptions and events regarding unrepresentable date times, so all code that handles that should be removed.

Here's a list of PRs that introduced the things that should be removed:
* #5023
* #5051
* #5235

### PR DESCRIPTION
**Pull request by sofokli1**
## What's this?
Remove all tracking regarding unrepresentable dates

### Relationships
Closes #5236

## Why do we want this?
looks like we don't need this anymore

## How is it done?
by 🔥

## Review hints
Take a look at the issues where it references the PRs that the code was initially added

## :squid: Permissions
Only me

### CODE REVIEW

File: Toggl.iOS/Views/EditDuration/WheelForegroundView.cs [hunk 1]
```diff
@@ -370,14 +370,7 @@ private void updateEditedTime(TimeSpan diff)
...
             {
-                var exceptionProperties = new Dictionary<string, string>
-                {
-                    { "DateTimeOffset", currentDateTimeOffset.ToString() },
-                    { "TimeSpan", difference.ToString() }
-                };
-                var exception = new UnrepresentableDateException("Unrepresentable DateTimeOffset encountered in WheelForegroundView", argumentOutOfRangeException);
-                IosDependencyContainer.Instance.AnalyticsService.Track(exception, exceptionProperties);
-                return currentDateTimeOffset;
+                throw argumentOutOfRangeException;
```
Comment (heytherewill) [inline Toggl.iOS/Views/EditDuration/WheelForegroundView.cs]: Please don't rethrow exceptions, as this empties the call stack. If we are removing all the logging, this means that the error is fixed, correct? We can safely remove all exceptions in this case

File: Toggl.Core.UI/Transformations/DateTimeToFormattedString.cs [hunk 2]
```diff
@@ -11,38 +11,27 @@ public class DateTimeToFormattedString
...
-                {
-                    { "DateTimeOffset", value.ToString() },
-                    { "TimeZoneName", timeZone.DisplayName },
-                    { "TimeZoneOffset", timeZone.BaseUtcOffset.ToString() }
-                };
-                var exception = new UnrepresentableDateException("Unrepresentable DateTimeOffset when converting to local timezone", argumentOutOfRangeException);
-                analyticsService.Track(exception, exceptionProperties);
-
-                throw exception;
+                throw argumentOutOfRangeException;
```
Comment (heytherewill) [inline Toggl.Core.UI/Transformations/DateTimeToFormattedString.cs]: Same thing about rethrow

File: Toggl.iOS/Views/EditDuration/WheelForegroundView.cs (hunk 1 above)
Comment (sofokli1) [reply Toggl.iOS/Views/EditDuration/WheelForegroundView.cs]: Maybe @arturs105 can comment if the error was fixed. I just know that the issue is in the current 2.1 release, so I am just executing the changes.

@arturs105 please review as well

@heytherewill will remove the rethrow

**Comment by arturs105** [comment]
I was gone on vacation for quite a while, so i'm a bit out of the loop on this. Did we analyse the tracked data and did we find where the out of range date times are coming from?

**Comment by sofokli1** [comment]
Personally was never in the loop about this. The issue says you added this to the 2.1 release via automation. @arturs105 who else was involved with this issue?

**Comment by arturs105** [comment]
😅  Yeah, i put this in 2.1 project, though as a rough estimate, since then it was 2 releases away and i estimated that by 2.1 we'd have gathered enough data and analysed it. I guess i should have been more explicit about first getting to the bottom of the Mysterious date times and only then removing the tracking.

Anyways it's not like this PR was wasted time, this is still usable, though a bit later - when we find out (and fix), the thing that is/was creating those date times. Putting this on hold for now.

**Comment by amulware** [comment]
@arturs105 	@sofokli1 	Status?

**Comment by sofokli1** [comment]
@amulware based on this [comment](https://github.com/toggl/mobileapp/issues/5237#issuecomment-510869551), this PR will be dealt with in iOS 2.3

**Comment by arturs105** [comment]
I went over crashes and events in iOS 2.1 - 2.1.3 and found nothing related to unrepresentable dates, so it's now time to move forward with this.

File: Toggl.iOS/Views/EditDuration/WheelForegroundView.cs [hunk 3]
```diff
@@ -363,23 +363,8 @@ private void updateEditedTime(TimeSpan diff)
         private DateTimeOffset getNewDateTimeOffset(
```
Comment (arturs105) [inline Toggl.iOS/Views/EditDuration/WheelForegroundView.cs]: This method isn't needed anymore. It was introduced to do safe addition with error catching, but now that error catching is removed and this method has become basically this (can be just simplified by using `+` operator):
```
public int sum(int a, int b)
    => a + b;
```

**Comment by sofokli1** [comment]
Not addressed yet

**Comment by sofokli1** [comment]
All addressed now

**Pull request merged**